Rocket.Chat/Controllers/Chat/ChatDataController.swift

```diff
@@ -32,107 +32,107 @@ struct ChatData {
     }
-    
+
     func itemAt(_ indexPath: IndexPath) -> ChatData? {
         for item in data {
             if item.indexPath?.row == indexPath.row && item.indexPath?.section == indexPath.section {
                 return item
             }
         }
-        
+
         return nil
     }
-    
+
     func indexPathOf(_ identifier: String) -> IndexPath? {
         for item in data {
             if item.identifier == identifier {
                 return item.indexPath
             }
         }
-        
+
         return nil
     }
-    
+
+    // swiftlint:disable function_body_length cyclomatic_complexity
     func insert(_ items: [ChatData]) -> [IndexPath] {
         var indexPaths: [IndexPath] = []
         var newItems: [ChatData] = []
         var lastObj = data.last
-        var identifiers: [String] = []
-        
+        var identifiers: [String] = items.map { $0.identifier }
+
         for obj in items {
             identifiers.append(obj.identifier)
         }
-        
+
         for newObj in items {
             if let lastObj = lastObj {
                 if lastObj.type == .message && (
                     lastObj.timestamp.day != newObj.timestamp.day ||
                     lastObj.timestamp.month != newObj.timestamp.month ||
                     lastObj.timestamp.year != newObj.timestamp.year) {
-                    
+
                     // Check if already contains some separator with this data
                     var insert = true
                     for obj in data.filter({ $0.type == .daySeparator }) {
-                        if (lastObj.timestamp.day == obj.timestamp.day &&
-                            lastObj.timestamp.month == obj.timestamp.month &&
-                            lastObj.timestamp.year == obj.timestamp.year) {
+                        if lastObj.timestamp.day == obj.timestamp.day &&
+                           lastObj.timestamp.month == obj.timestamp.month &&
+                           lastObj.timestamp.year == obj.timestamp.year {
                             insert = false
                         }
                     }
-                    
+
                     if insert {
+                        guard let calendar = NSCalendar(calendarIdentifier: .gregorian) else { continue }
                         let date = newObj.timestamp
-                        let calendar = NSCalendar(calendarIdentifier: .gregorian)!
-                        let components = calendar.components([.day , .month, .year ], from: date)
-                        let newDate = calendar.date(from: components)
-                        let separator = ChatData(type: .daySeparator, timestamp: newDate!)!
+                        let components = calendar.components([.day, .month, .year], from: date)
+                        guard let newDate = calendar.date(from: components) else { continue }
+                        guard let separator = ChatData(type: .daySeparator, timestamp: newDate) else { continue }
```

Comment:
If we call `continue` here, the new message will not be inserted in the list. Maybe let's insert this logic in a separated block, just to make sure that the new message get inserted?